astronomy/astronomy.F90

```diff
@@ -56,152 +51,37 @@ module astronomy_mod
-   module procedure diurnal_solar_2d
-   module procedure diurnal_solar_1d
-   module procedure diurnal_solar_0d
-   module procedure diurnal_solar_cal_2d
-   module procedure diurnal_solar_cal_1d
-   module procedure diurnal_solar_cal_0d
-end interface
-
-!> @brief Calculates the daily mean solar information for a given time and latitude.
-!!
-!> ~~~~~~~~~~{.f90}
-!! call daily_mean_solar (lat, time, cosz, fracday, rrsun)
-!! call daily_mean_solar (lat, time_since_ae, cosz, fracday, rrsun)
-!! call daily_mean_solar (lat, time, cosz, solar)
-!! call daily_mean_solar (lat, time_since_ae, cosz, solar)
-!! ~~~~~~~~~~
-!!
-!! The first option (used in conjunction with time_manager_mod)
-!! generates the real variable time_since_ae from the time_type
-!! input time, and then calls daily_mean_solar with this real input
-!! (option 2). The third and fourth options correspond to the first
-!! and second and are used with then spectral 2-layer model, where
-!! only cosz and solar are desired as output. These routines generate
-!! dummy arguments and then call option 2, where the calculation is done.
-!!
-!! The time of year is set by
-!! ~~~~~~~~~~{.f90}
-!!    real, intent(in) :: time_since_ae
-!! ~~~~~~~~~~
-!! With time_type input, the time of year is extracted from
-!! ~~~~~~~~~~{.f90}
-!!    type(time_type), intent(in) :: time
-!! ~~~~~~~~~~
-!!
-!! Separate routines exist within this interface for scalar,
-!! 1D or 2D input and output fields:
-!!
-!! ~~~~~~~~~~{.f90}
-!! real, intent(in), dimension(:,:) :: lat
-!! real, intent(in), dimension(:)   :: lat
-!! real, intent(in)                 :: lat
-!!
-!! real, intent(out), dimension(:,:) :: cosz, fracday
-!! real, intent(out), dimension(:)   :: cosz, fracday
-!! real, intent(out)                 :: cosz, fracday
-!! ~~~~~~~~~~
-!!
-!! @param [in] <lat> Latitudes of model grid points [radians]
-!! @param [in] <time_since_ae> Time of year; autumnal equinox = 0.0, one year = 2 * pi [radians]
-!! @param [in] <time> Time at which astronomical values are desired (time_type variable) [seconds, days]
-!! @param [out] <cosz> Cosine of solar zenith angle, set to zero when entire period is in darkness [dimensionless]
-!! @param [out] <fracday> Daylight fraction of time interval [dimensionless]
-!! @param [out] <rrsun> Earth-Sun distance (r) relative to semi-major axis of orbital ellipse
-!! (a):(a/r)**2 [dimensionless]
-!! @param [out] <solar> shortwave flux factor: cosine of zenith angle * daylight fraction /
-!! (earth-sun distance squared) [dimensionless]
-!> @ingroup astronomy_mod
+   module procedure diurnal_solar_2d_r4, diurnal_solar_2d_r8
+   module procedure diurnal_solar_1d_r4, diurnal_solar_1d_r8
+   module procedure diurnal_solar_0d_r4, diurnal_solar_0d_r8
+   module procedure diurnal_solar_cal_2d_r4, diurnal_solar_cal_2d_r8
+   module procedure diurnal_solar_cal_1d_r4, diurnal_solar_cal_1d_r8
+   module procedure diurnal_solar_cal_0d_r4, diurnal_solar_cal_0d_r8
+end interface diurnal_solar
+
+
```

Comment:
Should there be more interfaces for all the subroutines and functions in astronomy.inc?